src/main/fc/fc_msp.c

```diff
@@ -748,16 +748,30 @@ static bool mspCommonProcessOutCommand(uint8_t cmdMSP, sbuf_t *dst, mspPostProce
         sbufWriteU8(dst, batteryConfig()->currentMeterSource);
         break;
 
-    case MSP_TRANSPONDER_CONFIG:
+    case MSP_TRANSPONDER_CONFIG: {
 #ifdef TRANSPONDER
-        sbufWriteU8(dst, 1); //Transponder supported
-        for (unsigned int i = 0; i < sizeof(transponderConfig()->data); i++) {
-            sbufWriteU8(dst, transponderConfig()->data[i]);
+        sbufWriteU8(dst, TRANSPONDER_PROVIDER_COUNT);
```

Comment:
Ack.

I think I will be:
Backward compatible block
    provider count (1)
    ilap data
New bblock
    provider count (3)
    { provider id, required data length } x 3
    active provider id
    active provider dependent data